iothub/device/src/InternalClient.cs

```diff
@@ -284,14 +207,14 @@ public async Task OpenAsync()
         }
 
         /// <summary>
-        /// Explicitly open the InternalClient instance.
+        /// Explicitly open the client instance.
         /// </summary>
         /// <param name="cancellationToken">A token to cancel the operation.</param>
-        public Task OpenAsync(CancellationToken cancellationToken)
+        public async Task OpenAsync(CancellationToken cancellationToken)
         {
             try
             {
-                return InnerHandler.OpenAsync(cancellationToken);
+                await InnerHandler.OpenAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Fixing this bug where the try/catch didn't actually work once the callee went async.